### PR TITLE
Text_adventure_menu

### PR DESCRIPTION
**Pull request by DIONYSUS-BASOUNAS**
did my best

### CODE REVIEW

File: menu.h [hunk 1]
```diff
@@ -0,0 +1,103 @@
+#include <iostream>
+#include "utils.h"
+
+using namespace std;
```
Comment (achilleasmitos) [inline menu.h:4]: Delete the "using" line. Let's have explicit calls for the functions from the std lib.

File: menu.h [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+namespace menu
+{
+    void menu()
```
Comment (achilleasmitos) [inline menu.h:7]: I would name this function something more specific, like "start_menu" or something alike. There will be more "menus" down the line.

File: menu.h [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            case 2:
+            {
+                cout << "Pick Weight" << endl;
+                cout << "1. light" << endl;
+                cout << "2. SemiLight" << endl;
+                cout << "3. Normal" << endl;
+                cout << "4. SemiBold" << endl;
+                cout << "5. Bold" << endl;
+                cin >> weightsize;
+                if (weightsize == 1)
```
Comment (achilleasmitos) [inline menu.h:57]: Could also use `switch` statement here...

File: menu.h [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+namespace menu
+{
+    void menu()
+    {
+        cout << "Welcome to the game!" << endl;
+        cout << "Please select an option:" << endl;
+        cout << "1. Start Game" << endl;
+        cout << "2. Settings" << endl;
+        cout << "3. Quit" << endl;
+        int textsize=16;
```
Comment (achilleasmitos) [inline menu.h:14]: I believe we need to move `text_size` and `font_weight` outside of the function for this to work properly.

**Review by achilleasmitos — CHANGES_REQUESTED** [review]

File: menu.h
Comment (achilleasmitos) [inline menu.h:1]: Please also see https://github.com/achilleasmitos/text-adventure/pull/11 to fix the header/cpp includes directives. Header files should not have function definitions.

**Review by achilleasmitos — CHANGES_REQUESTED** [review]

File: menu.h [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                }
+                else if (weightsize == 4)
+                {
+                    utils::customize_text(textsize, utils::SemiBold);
+                }
+                else if (weightsize == 5)
+                {
+                    utils::customize_text(textsize, utils::Bold);
+                }
+                menu();
```
Comment (achilleasmitos) [inline menu.h:77]: This causes recursive behavior for the `menu()` function. While this shouldn't be a problem for our case, it isn't good programming practice. I believe we should come up with a way to ask the user for their choice (and follow through with what they choose) until a certain choice (exit) has been made. Something with a `do...while(...)` should do the trick here.

File: menu.h [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                break;
+            }
+            }
+        }
+        case 3:
+        {
+            // code for quitting
+            cout << "Quitting the game..." << endl;
+            exit(0);
```
Comment (achilleasmitos) [inline menu.h:92]: This function does not do proper cleanup (in particular, it does not call the destructor of any automatic-storage object), and it shouldn't be used. The proper way to terminate the program is to `return 0` from `main()`. Read more here: https://stackoverflow.com/questions/30250934/how-to-end-c-code

**Review by achilleasmitos — CHANGES_REQUESTED** [review]

File: menu.h (hunk 2 above)
Comment (DIONYSUS-BASOUNAS) [reply menu.h:7]: done

File: menu.h (hunk 3 above)
Comment (DIONYSUS-BASOUNAS) [reply menu.h:57]: you we could by i like if more :D

File: menu.h (hunk 4 above)
Comment (DIONYSUS-BASOUNAS) [reply menu.h:14]: done

File: menu.h (hunk 6 above)
Comment (DIONYSUS-BASOUNAS) [reply menu.h:92]: finaly did this one too

File: menu.h (hunk 5 above)
Comment (DIONYSUS-BASOUNAS) [reply menu.h:77]: i think i got it

File: menu.h
Comment (DIONYSUS-BASOUNAS) [reply menu.h:1]: i don't get it

File: .vscode/settings.json [hunk 7]
```diff
@@ -55,6 +55,29 @@
 		"streambuf": "cpp",
 		"thread": "cpp",
 		"typeinfo": "cpp",
-		"future": "cpp"
+		"future": "cpp",
```
[the remark this answers is not in the document]
Comment (DIONYSUS-BASOUNAS) [reply .vscode/settings.json:58]: ok

File: .vscode/settings.json [hunk 8]
```diff
@@ -55,6 +55,6 @@
 		"streambuf": "cpp",
 		"thread": "cpp",
 		"typeinfo": "cpp",
-		"future": "cpp"
+
```
Comment (achilleasmitos) [inline .vscode/settings.json:58]: "future" was in the file before. please leave it there

Note, that I do not know where exactly this settings.json file is used (I only have a faint idea). I will look into it for later development.

File: utils.h [hunk 9]
```diff
@@ -10,7 +10,8 @@ namespace utils
 {
     // Number of milliseconds that the 'print' function waits for in-between character prints.
     int sleep_for_ms = 100;
-
+    int textsize=16;
+    int weightsize=400;
```
Comment (achilleasmitos) [inline utils.h:14]: 'weightsize' is not a good name for this variable. also, this shouldn't be an integer, but rather a FontWeightValues enum

Comment (achilleasmitos) [reply utils.h:14]: Please see the naming convention followed in the rest of the files. change `textsize` to `text_size` (similar for other stuff.

File: start_menu.h [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+        std::cout << "Welcome to the game!" << std::endl;
+        int choice = -1;
+
+        while (choice != 1 ) {
+        std::cout << "Please select an option:" << std::endl;
+        std::cout << "1. Start Game" << std::endl;
+        std::cout << "2. Settings" << std::endl;
+        std::cout << "3. Quit" << std::endl;
+        
+        int choice;
```
Comment (achilleasmitos) [inline start_menu.h:17]: redefinition of `choice`

File: start_menu.h [hunk 11]
```diff
@@ -0,0 +1,109 @@
+#include <iostream>
+#include "utils.h"
+
+namespace menu
```
Comment (achilleasmitos) [inline start_menu.h:4]: change the namespace to `start_menu` as well

File: start_menu.h [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+                break;
+            }
+            }
+            break;
+        }
+        case 3:
+        {
+            // code for quitting
+            std::cout << "Quitting the game..." << std::endl;
+            return 1;
```
Comment (achilleasmitos) [inline start_menu.h:94]: `return 1;` indicates a mistake. we want to exit this function and return to `main()`, where we will `return 0;` as well, and exit the program

**Review by achilleasmitos — CHANGES_REQUESTED** [review]

File: menu.h
Comment (achilleasmitos) [reply menu.h:1]: We'll take this offline, since it is quite the conversation :)

File: start_menu.h (hunk 11 above)
Comment (DIONYSUS-BASOUNAS) [reply start_menu.h:4]: done

File: .vscode/settings.json (hunk 8 above)
Comment (DIONYSUS-BASOUNAS) [reply .vscode/settings.json:58]: done

**Review by achilleasmitos — APPROVED** [review]

**Pull request merged**